### PR TITLE
Fullpipe Common::String + memory leak fixes

### PR DESCRIPTION
**Pull request by bluegr**
In this pull request, I've changed a lot of engine strings in Fullpipe so that they are handled via Common::String. This results in cleaner and more manageable code.

I've tried splitting up the commits so that all relevant changes are grouped together

Update: I've also added some memory leak fixes to this branch, as part of fixing memory leaks in this engine. These are relevant, because the original intention of this branch was to optimize code and identify memory leaks.

### CODE REVIEW

File: engines/fullpipe/scene.cpp [hunk 1]
```diff
@@ -221,10 +210,10 @@ bool Scene::load(MfcArchive &file) {
 
 	_libHandle = g_fp->_currArchive;
 
-	if (_picObjList.size() > 0 && _bgname && strlen(_bgname) > 1) {
+	if (_picObjList.size() > 0 && _bgname.size() > 1) {
```
Comment (sev-) [inline engines/fullpipe/scene.cpp]: !_bgname.empty()

Comment (bluegr) [inline engines/fullpipe/scene.cpp]: Fixed in af874d4

**Comment by sev-** [comment]
Sweet.

**Comment by wjp** [comment]
Looks good, but I don't understand why the Common::String function arguments are a mix of by-value and by-(non-const-)reference, instead of the standard `const Common::String &`.

**Comment by bluegr** [comment]
Good point @wjp. I'll adapt these accordingly on master

**Pull request merged**